RocketChat/Rocket.Chat
# Regression: Update team's record when its main channel is updated

## Proposed changes (including videos or screenshots)


 - Add `name` and `type` update to the team's record in the `saveRoomSettings` method.



## Issue(s)

[Task - ClickUp](https://app.clickup.com/t/452jt2)

## Steps to test or reproduce

Enter a team's main room > Team's Information > Edit. Edit the team's main channel's name and type (private or public) and Save.
The corresponding team's record should be updated in the database after this process. 

## Further comments

This PR should be merged after #21134
